tools/slang-test/main.cpp

```diff
@@ -1745,12 +1745,88 @@ void runTestsInDirectory(
     }
 }
 
+struct ToolInvoke
+{
+        /// Returns true if it is a tool invoke
+    bool parse(const char*const* argv, int argc)
+    {
+        m_args.Clear();
+
+        m_binDirectory = ".";
+
+        if (argc < 2 || !_isToolName(argv[1]))
+        {
+            return false;
+        }
+        m_toolName = argv[1];
+
+        // Look for parameters that are for the slang-test, and should be skipped
+        int i = 2;
```

Comment:
Can we have a model for subcommands closer to `git`, where the options for the "driver" come before the tool name, and then tool-specific options come after? Something like `slang-test [options for slang-test] tool-name [options for tool]`.

Reply:
So this was just a first stab at this to get something going such that it was backwards compatible. If we don't mind breaking the command line for slang-test a bit, to make it easy to identify the 'subcommand' then this would make this better.